File: code/java/src/main/java/state/TrafficLight.java
package state;

class TrafficLight {
    private State currentState;

    public TrafficLight(State state) {
        this.currentState = state;
    }

    public void perform() {
        currentState.handleRequest();
        // Logic to change the current state
    }

    public void setState(State state) {
        this.currentState = state;
    }
}

File: code/typescript/state/TrafficLight.ts
import { IState } from "./State";

export class TrafficLight {
    private currentState: IState;

    constructor(state: IState) {
        this.currentState = state;
    }

    perform(): void {
        this.currentState.handleRequest();
        // Logic to change the current state
    }

    setState(state: IState): void {
        this.currentState = state;
    }
}
